pkg/spoke/addon/registration_controller.go

```diff
@@ -175,7 +178,15 @@ func (c *addOnRegistrationController) syncAddOn(ctx context.Context, syncCtx fac
 // startRegistration starts a client certificate controller with the given config
 func (c *addOnRegistrationController) startRegistration(ctx context.Context, config registrationConfig) context.CancelFunc {
 	ctx, stopFunc := context.WithCancel(ctx)
-	kubeInformerFactory := informers.NewSharedInformerFactoryWithOptions(c.spokeKubeClient, 10*time.Minute, informers.WithNamespace(config.installationNamespace))
+
+	var kubeClient kubernetes.Interface = c.spokeKubeClient
```

Comment:
I would call it managementKubeClient here, It needs some explanation on why not directly use managedKubeClient.